shell/src/main/java/alluxio/cli/fs/command/LoadCommand.java

```diff
@@ -51,31 +62,46 @@ public String getCommandName() {
   }
 
   @Override
+  public Options getOptions() {
+    return new Options()
+        .addOption(LOCAL_OPTION);
+  }
+
+  @Override
   protected void runCommand(AlluxioURI path, CommandLine cl) throws AlluxioException, IOException {
-    load(path);
+    load(path, cl.hasOption(LOCAL_OPTION.getLongOpt()));
   }
 
   /**
-   * Loads a file or directory in Alluxio space, makes it resident in memory.
+   * Loads a file or directory in Alluxio space, makes it resident in Alluxio.
    *
-   * @param filePath The {@link AlluxioURI} path to load into Alluxio memory
+   * @param filePath The {@link AlluxioURI} path to load into Alluxio
+   * @param local whether to load data to local worker even when the data is already loaded remotely
    */
-  private void load(AlluxioURI filePath) throws AlluxioException, IOException {
+  private void load(AlluxioURI filePath, boolean local) throws AlluxioException, IOException {
     URIStatus status = mFileSystem.getStatus(filePath);
     if (status.isFolder()) {
       List<URIStatus> statuses = mFileSystem.listStatus(filePath);
       for (URIStatus uriStatus : statuses) {
         AlluxioURI newPath = new AlluxioURI(uriStatus.getPath());
-        load(newPath);
+        load(newPath, local);
       }
     } else {
-      if (status.getInAlluxioPercentage() == 100) {
-        // The file has already been fully loaded into Alluxio memory.
+      OpenFileOptions options = OpenFileOptions.defaults().setReadType(ReadType.CACHE_PROMOTE);
+      if (local && !(FileSystemContext.INSTANCE.hasLocalWorker()
+                   && options.getCacheLocationPolicy() instanceof LocalFirstPolicy)) {
```

Comment:
what do you think of explicitly setting the cache location policy to `LocalFirstPolicy` when `--local` is used?

Reply:
@aaudiber really better